dfp/decimal/decimal

```diff
@@ -180,6 +180,286 @@ public:
+	  __err = ios_base::failbit;
+	}
+
+      setlocale(LC_ALL, __sav);
+      delete [] __sav;
+    }
+
+
+
+  template<typename _CharT, typename _InIter = std::istreambuf_iterator<_CharT>>
+  class dfp_num_get : public num_get<_CharT, _InIter> {
+      typedef _InIter iter_type;
+
+  protected:
+      virtual iter_type
+      do_get(iter_type, iter_type, ios_base&, ios_base::iostate&,
+        decimal::decimal32&) const;
+
+      virtual iter_type
+      do_get(iter_type, iter_type, ios_base&, ios_base::iostate&,
+        decimal::decimal64&) const;
+
+      virtual iter_type
+      do_get(iter_type, iter_type, ios_base&, ios_base::iostate&,
+        decimal::decimal128&) const;
+
+  public:
+      iter_type
+      get(iter_type in, iter_type end, ios_base& io,
+     ios_base::iostate& err, decimal::decimal32& v) const
+      { return this->do_get(in, end, io, err, v); }
+
+      iter_type
+      get(iter_type in, iter_type end, ios_base& io,
+          ios_base::iostate& err, decimal::decimal64& v) const
+      { return this->do_get(in, end, io, err, v); }
+
+      iter_type
+      get(iter_type in, iter_type end, ios_base& io,
+          ios_base::iostate& err, decimal::decimal128& v) const
+      { return this->do_get(in, end, io, err, v); }
+  };
+
+  template<typename _CharT, typename _InIter>
+    _InIter
+    dfp_num_get<_CharT, _InIter>::
+    do_get(iter_type __beg, iter_type __end, ios_base& __io,
+           ios_base::iostate& __err, decimal::decimal32& __v) const
+    {
+      string __xtrc;
+      __xtrc.reserve(32);
+      __beg = this->_M_extract_float(__beg, __end, __io, __err, __xtrc);
+      std::__convert_to_v(__xtrc.c_str(), __v, __err, locale::facet::_S_get_c_locale());
+      if (__beg == __end)
+        __err |= ios_base::eofbit;
+      return __beg;
+    }
+
+  template<typename _CharT, typename _InIter>
+    _InIter
+    dfp_num_get<_CharT, _InIter>::
+    do_get(iter_type __beg, iter_type __end, ios_base& __io,
+           ios_base::iostate& __err, decimal::decimal64& __v) const
+    {
+      string __xtrc;
+      __xtrc.reserve(32);
+      __beg = this->_M_extract_float(__beg, __end, __io, __err, __xtrc);
+      std::__convert_to_v(__xtrc.c_str(), __v, __err, locale::facet::_S_get_c_locale());
+      if (__beg == __end)
+        __err |= ios_base::eofbit;
+      return __beg;
+    }
+
+  template<typename _CharT, typename _InIter>
+    _InIter
+    dfp_num_get<_CharT, _InIter>::
+    do_get(iter_type __beg, iter_type __end, ios_base& __io,
+           ios_base::iostate& __err, decimal::decimal128& __v) const
+    {
+      string __xtrc;
+      __xtrc.reserve(32);
+      __beg = this->_M_extract_float(__beg, __end, __io, __err, __xtrc);
+      std::__convert_to_v(__xtrc.c_str(), __v, __err, locale::facet::_S_get_c_locale());
+      if (__beg == __end)
+        __err |= ios_base::eofbit;
+      return __beg;
+    }
+
+  template <class charT, class traits = char_traits<charT>>
+  class dfp_basic_istream;
```

Comment:
Why add new istream types? That defeats the entire purpose of using istreams, which is that all istream types provide the same API and you don't need to care whether you're using `ifstream` or `istringstream` etc.

If users can only extract decimal types when using `dfp_basic_istream` then how do they interact with `ifstream` etc?

 This is **not** how you extend iostreams for custom types.

Also, `basic_dfp_istream` would be more consistent with naming conventions. But this type should just not exist at all.